fairroot/geobase/FairGeoSet.h

```diff
@@ -89,10 +90,21 @@ class FairGeoSet : public TNamed
     TString& getAuthor() { return author; }
     TString& getDescription() { return description; }
     virtual const char* getKeepinName(Int_t, Int_t) { return 0; }
-    virtual const char* getModuleName(Int_t) { return 0; }
-    virtual const char* getEleName(Int_t) { return 0; }
+    virtual const char* getModuleName(Int_t);
+    virtual const char* getEleName(Int_t);
```

Comment:
Actually, thinking more about those functions: I think we should go one step further and change the return type and return a string (by-value). Returning a `const char *` is really dangerous, because those functions look like ordinary getters: all results of a previous call become dangling pointers as soon as the next call happens.

Reply:
I thought about the same but unfortunately these functions are public and definitely used in user code so we can't change it without breaking the user code. 
Is it possible to implement a second function which only has a different return type? If so we can deprecate the old function and internally use the new one.

Reply:
Yes, that would be the better approach for a smoother transition.

Reply:
We probably should start by seperating the interfaces that should be called and that should be overriden (there's some cool "pattern" name for this, I have forgotten it).

We should deprecate `getModuleName` so that people stop calling it.

And here's an idea for the "to be called" method.

```cpp
std::string CallGetModuleName(Int_t x)
{
    // getModuleName usually is not thread safe!
    std::mutex mtx;
    std::lock_guard lock{mtx};

#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
    return {getModuleName(x)};
#pragma GCC diagnostic pop
}
```

Reply:
Good idea but we should try to get this PR merged and create issues with the still open problems/suggestions. 
We should now focus on getting the warnings fixed such that we can create the new FairRoot release.